### ISSUE TITLE
Protocol vs Delegate

### ISSUE THREAD

**Issue by massimobio** [celanajaya/AdjustableTextField#3]
In response to @celanajaya 's question in the comments on the AdjustableTextField, I like the idea of having both options.
Maybe we should have two controls in the demo app, one using delegation and one using the closure.

As far as naming I was thinking that onValueChanged sounds a little Androidy to me.

What do you think about valueChangedHandler and didChangeValue for the Delegate Protocol?
If there was use for it we could even add more delegate methods such as didStartDragging and didFinishDragging in case we wanted to updated to UI or wait for the final value to be ready before acting on it... possibilities

**Comment by massimobio** [comment]
You added a second example so we cover both methods now in #5 👍

Sorry If I keep going on about naming stuff but I feel it's important since this might be used by others.

I made pull requests for the remaining naming issues I listed above
closure name #9 
delegate method name #10 

we can close this since all issues are now addressed in PRs.

Come to think of it, it seems to me like many of the issues should rather just be PRs directly without the duplicating effort of writing an issue first. If needed, discussions can happen in the PR's comments. I'll try to steer in that direction from now on.
